### ISSUE TITLE
Function omissions

### ISSUE THREAD

**Issue by jklmli** [amacdougall/underscore.as#1]
No `_.union, _.intersection, _.difference, _.groupBy` functions from 1.1.7.
No `_.defaults` function from 1.1.5.

Were these intentional, or should I submit a patch?

**Comment by amacdougall** [comment]
Hey, sorry I didn't notice this update! We use Github at work, and notification spam made it hard to follow things. I hope that the new notification system will make that less of a problem from now on.

This library was based on an earlier version of Underscore, and I've only been adding features when I need them, which isn't often. I would definitely appreciate a patch, if you've added those functions yourself!

**Comment by jklmli** [comment]
Sure, I'll attach it once I get around to writing some tests for them.

**Comment by amacdougall** [comment]
`_.intersection` was in there, it just had the incorrect name `_.intersect`. I've changed the name, and kept the original wrong name as an alias. @DrBenton has been kind enough to write an implementation of `_.difference`, which is merged in. This leaves `_.union`, `_.groupBy`, and `_.defaults`.

**Comment by mhulse** [comment]
I know this issue is old ... anyone figure out `union`?

**Comment by mhulse** [comment]
> I know this issue is old ... anyone figure out `union`?

I should have looked harder before asking. Found some good examples via:

https://searchcode.com/?q=union&loc=0&loc2=10000&lan=42

Like:

https://github.com/drewbourne/asx

Anyway, thanks for this awesome bit of code.
